### PR TITLE
Second view

### PR DESCRIPTION
**Pull request by josephmcgovern-wf**
### What was done
- Split js into it's own files:
- `game.js` (game object)
- `player.js` (player object)
- `home.js` (controller)
- Now fetch users from the datastore and select from them (there's a bug here that needs to be fixed - if you select a game-type too fast, no users will show up. This is because we don't wait for the ajax request before creating the select2. I'll probably use a `setInterval` or something to solve this in my next pr).
- Now can fully fill out user data and send data to a game api endpoint to be put into datastore.
- I decided to put API files into their given directory (user api goes into /user/ folder). I'm ok with switching it, but I kind of like it, so we don't have duplicate `user.py` files. Obviously, we have duplicate `api` files, but I think it's easier to fuzzy find them (`user api` instead of `api user`). Just thought we'd try it out and see how it feels.
### What still needs to be done
- [x] Add feedback (thinking about taking our feedback class)
- [ ] You and I need to talk about defaulting some of these values.
- [x] Add forward/back buttons to visit previous/next players. 
- [x] Don't allow a single-person game 
- [ ] `master_urls` file? 
- [x] No players appear bug

So, the front end is now functional. Needs some polish work, but you should definitely be able to add games to the datastore now using our UI.

EDIT:
I merged in some of your changes to successfully put some stuff into the datastore
Screencap: http://screencast.com/t/tpvAVWG3E0DR

### CODE REVIEW

**Comment by travisreed-wf** [comment]
I am ok with this, but I think that having api functions in an api folder makes more sense. Think of how often you are looking for a specific api endpoint and arent quite sure which of 3 possible files it could be in. Its a lot easier to flip through api files that are next to each other, rather than api files in multiple modules.

However, like I said, I am ok with giving this a shot, just wanted to provide my commentary.

File: templates/js/home.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  });
+});
+
+
+function fetchUsernames() {
+  $.ajax({
+    url: '/api/user/usernames/',
+    method: 'GET',
+    success: function(data) {
+      USERNAMES = JSON.parse(data);
```
Comment (travisreed-wf) [inline templates/js/home.js]: why caps?

File: templates/js/player.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        '<div class="form-group">' +
+          '<label>Lag rank</label>' +
+          '<input id="lag-rank" class="form-control" type="number">' +
+        '</div>' +
+        '<button id="player-submit" class="btn btn-primary">Submit</button>' +
+      '</div>';
+    return html;
+  },
+
+  getInputData: function() {
```
Comment (travisreed-wf) [inline templates/js/player.js]: parseInputs?

Comment (josephmcgovern-wf) [inline templates/js/player.js]: +1

File: templates/js/home.js (hunk 1 above)
Comment (josephmcgovern-wf) [inline templates/js/home.js]: It's a global that's accessed outside of this file (in `game.js`)

**Comment by josephmcgovern-wf** [comment]
travis - I moved the api files to their own folder. I didn't feel super strongly about it.

File: templates/js/home.js (hunk 1 above)
Comment (travisreed-wf) [inline templates/js/home.js]: but it isnt a constant, its coming from the server.

File: src/game/pool_game.py [hunk 3]
```diff
@@ -7,6 +7,19 @@
 class PoolGame(Game):
     """Represents a game of Pool"""
 
+    def calculate_rank_points_changes(self):
```
Comment (travisreed-wf) [inline src/game/pool_game.py]: I dont think you meant to make this change.

**Pull request merged**